core/src/main/scala/org/apache/spark/util/Executable.scala

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.util
+
+/*
+ * Adds main method to a class or object
+ */
+private[spark] trait Executable {
```

Comment:
Ok so in the code there is no reason as you said. In the test code I tried to re-use this function [here]:
https://github.com/skonto/spark/blob/d40b928aba0f44c5b3397d61d4450489e5d387c6/core/src/test/scala/org/apache/spark/deploy/SparkSubmitSuite.scala#L61
and I need to pass an object for testing and that object needs to implement some specific functions.
Spark project does not allow duck typing and I thought reflection is too much. Anyway I can eliminate it with reflection. I could have avoided in the first place but not reusing the function but I thought I could improve test code too...